src/dash-table/components/ControlledTable/TooltipCursorDecorator.tsx

```diff
@@ -0,0 +1,201 @@
+import * as R from 'ramda';
+import React, { PureComponent } from 'react';
+
+import DOM from 'core/browser/DOM';
+
+import { ColumnId, IVirtualizedDerivedData, ITableStaticTooltips, ITableTooltips } from '../Table/props';
+import { Tooltip, ConditionalTooltip, TooltipSyntax } from 'dash-table/tooltips/props';
+import SyntaxTree from 'core/syntax-tree';
+
+interface IProps {
+    column_static_tooltip: ITableStaticTooltips;
+    column_conditional_tooltips: ConditionalTooltip[];
+    delay: number;
+    tooltips?: ITableTooltips;
+    virtualized: IVirtualizedDerivedData;
+}
+
+interface IState {
+    columnId?: ColumnId;
+    md?: Remarkable;
+    displayTooltip?: boolean;
+    pageX?: number;
+    pageY?: number;
+    rowIndex?: number;
+    tooltipTimeoutId?: any;
+}
+
+export default class TooltipCursorDecorator extends PureComponent<IProps, IState> {
+    constructor(props: IProps) {
+        super(props);
+
+        this.state = {};
+    }
+
+    private getTooltip(): Tooltip | undefined {
+        const { pageX, pageY, columnId, rowIndex } = this.state;
+        const { tooltips, column_conditional_tooltips, column_static_tooltip } = this.props;
+
+        if (pageX === undefined || pageY === undefined || columnId === undefined || rowIndex === undefined) {
+            return undefined;
+        }
+
+        const { virtualized } = this.props;
+
+        const realIndex = rowIndex + virtualized.offset.rows;
+
+        const legacyTooltip = tooltips &&
+            tooltips[columnId] &&
+            (
+                tooltips[columnId].length > realIndex ?
+                    tooltips[columnId][realIndex] :
+                    null
+            );
+
+        const staticTooltip = column_static_tooltip[columnId];
+
+        const conditionalTooltips = R.filter(tt => {
+            return !tt.if ||
+                (
+                    (
+                        tt.if.column_id === undefined ||
+                        tt.if.column_id === columnId
+                    ) &&
+                    (
+                        tt.if.row_index === undefined ||
+                        tt.if.row_index === realIndex ||
+                        (tt.if.row_index === 'odd' && realIndex % 2 === 1) ||
+                        (tt.if.row_index === 'even' && realIndex % 2 === 0)
+                    ) &&
+                    (
+                        tt.if.filter === undefined ||
+                        (() => {
+                            const ast = new SyntaxTree(tt.if.filter);
+                            return ast.isValid && ast.evaluate(
+                                this.props.virtualized.data[realIndex - this.props.virtualized.offset.rows]
+                            );
+                        })()
+                    )
+                );
+        }, column_conditional_tooltips);
+
+        return conditionalTooltips.length ?
+            conditionalTooltips[0] :
+            legacyTooltip || staticTooltip;
+    }
```

Comment:
This `if` block of logic is very closely related to another one for the filters -- need to refactor / cleanup usage